core/dbt/graph/selector.py

```diff
@@ -135,30 +136,57 @@ def __init__(self, linker, manifest):
         self.linker = linker
         self.manifest = manifest
 
+    def _node_iterator(self, graph, exclude, include):
+        for node in graph.nodes():
+            real_node = self.manifest.nodes[node]
+            if include is not None and real_node.resource_type not in include:
+                continue
+            if exclude is not None and real_node.resource_type in exclude:
+                continue
+            yield node, real_node
+
+    def parsed_nodes(self, graph):
+        return self._node_iterator(
+            graph,
+            exclude=(NodeType.Source,),
+            include=None)
+
+    def source_nodes(self, graph):
+        return self._node_iterator(
+            graph,
+            exclude=None,
+            include=(NodeType.Source,))
+
     def get_nodes_by_qualified_name(self, graph, qualified_name_selector):
         """Yield all nodes in the graph that match the qualified_name_selector.
 
         :param str qualified_name_selector: The selector or node name
         """
         qualified_name = qualified_name_selector.split(".")
         package_names = get_package_names(graph)
-        for node in graph.nodes():
-            real_node = self.manifest.nodes[node]
-            if real_node.resource_type == NodeType.Source:
-                continue
-            fqn_ish = real_node.fqn
-            if _node_is_match(qualified_name, package_names, fqn_ish):
+        for node, real_node in self.parsed_nodes(graph):
+            if _node_is_match(qualified_name, package_names, real_node.fqn):
                 yield node
 
     def get_nodes_by_tag(self, graph, tag_name):
         """ yields nodes from graph that have the specified tag """
-        for node in graph.nodes():
-            real_node = self.manifest.nodes[node]
-            if real_node.resource_type == NodeType.Source:
-                continue
-            tags = real_node.tags
+        for node, real_node in self.parsed_nodes(graph):
+            if tag_name in real_node.tags:
+                yield node
 
-            if tag_name in tags:
+    def get_nodes_by_source(self, graph, source_full_name):
+        """yields nodes from graph are the specified source."""
+        if source_full_name.count('.') != 1:
```

Comment:
Let's support selection by source name (eg. `source:snowplow`) as well as table name (eg. `source:snowplow.event`) here

Reply:
@drewbanin I've implemented this (ignore appveyor, just the usual timeouts)